ProcessList.c

```diff
@@ -370,6 +370,16 @@ void ProcessList_collapseAllBranches(ProcessList* this) {
    }
 }
 
+static void ProcessList_filterChildren(ProcessList *this, pid_t pid, Hashtable *processFilter) {
+   for (int i = Vector_size(this->processes) - 1; i >= 0; i--) {
+      Process *p = (Process*) (Vector_get(this->processes, i));
+      if (p->pid != pid && Process_isChildOf(p, pid)) {
+         Hashtable_put(processFilter, p->pid, (void*) 1);
+         ProcessList_filterChildren(this, p->pid, processFilter);
```

Comment:
Do you have an idea how to do better?

Reply:
Even a single loop over processes with walking up from every process would be better (although it would need some cycle detection too) as it can easily be made worst case linear in the total process number. I personally think something like the tree mode "sort by parent" would be even better, but I'm a bit out of context at the moment and not sure how feasible it would be to share that part instead of reimplementing it.

Reply:
You could make the hit detection (in subtree/any parent) in O(1) if we tracked some item ranges for nested sets (similar to what Joomla is doing for its menus). Unfortunately, when we tried to implement this in-place while building the tree, this made tree creation kinda slow (`O(n²)`). One option could be to calculate the nested set ranges on the fly from the `Table->displayList` based on the indent columns; takes at most `O(n*m)` with `m` being the maximum tree depth (which is fixed at 32).